roxie/ccd/ccdprotocol.cpp

```diff
@@ -1523,12 +1522,13 @@ class RoxieSocketWorker : public ProtocolQueryWorker
     //  interface IPooledThread
     virtual void init(void *_r) override
     {
-        client.setown(new CSafeSocket((ISocket *) _r));
+        rawClient.setown((ISocket *) _r);
         ProtocolQueryWorker::init(_r);
     }
 
     virtual void threadmain() override
     {
+        client.setown(new CSafeSocket(listener->createSecureSocket(rawClient.getClear())));
```

Comment:
I think createSecureSocket() does not throw, should we check here if client is nullptr and avoid the doMain() call ?

Reply:
Yes.  client will be non null, but the result of createSecureSocket may be null.

Reply:
Owned<ISocket> clisock = listener->createSecureSocket(rawClient.getClear());
        if (clisock)
        {
            client.setown(new CSafeSocket(clisock.getClear()));
            doMain("");
        }
        else
            rawClient->shutdownNoThrow();

Reply:
Shutdown rawClient to help other side see/feel this ?